src/test/java/io/pinecone/clients/ConnectionsMapTest.java

```diff
@@ -0,0 +1,118 @@
+package io.pinecone.clients;
+
+import io.pinecone.configs.PineconeConfig;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+import org.openapitools.client.ApiException;
+import org.openapitools.client.api.ManageIndexesApi;
+import org.openapitools.client.model.IndexModel;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.Mockito.*;
+import static org.junit.jupiter.api.Assertions.*;
+
+public class ConnectionsMapTest {
+
+    @Test
+    public void testMultiplePineconeInstancesForIndexes() {
+        Pinecone pinecone1 = mock(Pinecone.class);
+        Pinecone pinecone2 = mock(Pinecone.class);
+
+        assertNotSame(pinecone1, pinecone2);
+
+        // Tests for blocking indexes
+        Index index1 = mock(Index.class);
+        Index index2 = mock(Index.class);
```

Comment:
Maybe this is me being unfamiliar with mockito, but I'm confused what these tests are doing if we're just mocking `Pinecone` which contains the `ConcurrentHashMap`. Aren't we just stubbing the behavior now instead of testing the internals of the class?

Reply:
Updated the tests